python/test/space/test_spatial_state.py

```diff
@@ -0,0 +1,45 @@
+import unittest
+
+from state_representation import SpatialState, StateType
+
+from test.test_state import STATE_METHOD_EXPECTS
+
+SPATIAL_STATE_METHOD_EXPECTS = [
+    'get_reference_frame',
+    'set_reference_frame',
+    'is_compatible'
+]
+
+
+class TestState(unittest.TestCase):
+
+    def test_callable_methods(self):
+        methods = [m for m in dir(SpatialState) if callable(getattr(SpatialState, m))]
+        for expected in STATE_METHOD_EXPECTS:
+            self.assertIn(expected, methods)
+        for expected in SPATIAL_STATE_METHOD_EXPECTS:
+            self.assertIn(expected, methods)
+
+    def test_constructors(self):
+        state1 = SpatialState(StateType.JOINTSTATE)
+        self.assertEqual(state1.get_type(), StateType.JOINTSTATE)
+        self.assertEqual(state1.get_name(), "none")
```

Comment:
Only by virtue of this test am I finding out that the default name for a `State` is `"none"`... I had no idea! I assumed it was the empty string. Most State-derived objects force you to supply a name or else are empty, so printing the name rarely happens. 

I'm actually not sure how I feel about it haha, I think I would honestly prefer an empty string rather than a specific keyword to indicate none-ness. Not technically the point of this PR but you've had to hard-code `"none"` in a bunch of tests now, so if we would change it I would suggest changing it now.

@buschbapti how would you feel if we changed lines 5 and 7 of State.cpp from the original:
```c++
State::State() : type_(StateType::STATE), name_("none"), empty_(true) {}

State::State(const StateType& type) : type_(type), name_("none"), empty_(true) {}
```

To explicitly empty:
```c++
State::State() : type_(StateType::STATE), name_(""), empty_(true) {}

State::State(const StateType& type) : type_(type), name_(""), empty_(true) {}
```

Reply:
I think this a good change. I don't think it would break any behavior down the line because we have empty checks and such.